rafay/resource_aks_cluster_v3.go

```diff
@@ -0,0 +1,3116 @@
+	for i := range p {
+		obj := &infrapb.Userassignedidentityexceptions{}
+		in := p[i].(map[string]interface{})
+
+		if v, ok := in["name"].(string); ok && len(v) > 0 {
+			obj.Name = v
+		}
+
+		if v, ok := in["namespace"].(string); ok && len(v) > 0 {
+			obj.Namespace = v
+		}
+
+		if v, ok := in["pod_labels"].(map[string]interface{}); ok {
+			obj.PodLabels = toMapString(v)
+		}
+		out[i] = obj
+	}
+	return out
+}
+
+func expandAKSV3ManagedClusterPrivateLinkResources(p []interface{}) []*infrapb.Privatelinkresources {
+	out := make([]*infrapb.Privatelinkresources, len(p))
+	if len(p) == 0 || p[0] == nil {
+		return out
+	}
+
+	for i := range p {
+		obj := &infrapb.Privatelinkresources{}
+		in := p[i].(map[string]interface{})
+		if v, ok := in["group_id"].(string); ok && len(v) > 0 {
+			obj.GroupId = v
+		}
+
+		if v, ok := in["id"].(string); ok && len(v) > 0 {
+			obj.Id = v
+		}
+
+		if v, ok := in["name"].(string); ok && len(v) > 0 {
+			obj.Name = v
+		}
+
+		if v, ok := in["required_members"].([]interface{}); ok && len(v) > 0 {
+			obj.RequiredMembers = toArrayString(v)
+		}
+
+		if v, ok := in["type"].(string); ok && len(v) > 0 {
+			obj.Type = v
+		}
+
+		out[i] = obj
+	}
+
+	return out
+}
+
+func expandAKSManagedClusterV3ServicePrincipalProfile(p []interface{}) *infrapb.Serviceprincipalprofile {
+	obj := &infrapb.Serviceprincipalprofile{}
+	if len(p) == 0 || p[0] == nil {
+		return obj
+	}
+	in := p[0].(map[string]interface{})
+
+	if v, ok := in["client_id"].(string); ok && len(v) > 0 {
+		obj.ClientId = v
+	}
+
+	if v, ok := in["secret"].(string); ok && len(v) > 0 {
+		obj.Secret = v
+	}
+
+	return obj
+}
+
+func expandAKSManagedClusterV3WindowsProfile(p []interface{}) *infrapb.Windowsprofile {
+	obj := &infrapb.Windowsprofile{}
+	if len(p) == 0 || p[0] == nil {
+		return obj
+	}
+	in := p[0].(map[string]interface{})
+
+	if v, ok := in["admin_username"].(string); ok && len(v) > 0 {
+		obj.AdminUsername = v
+	}
+
+	if v, ok := in["enable_csi_proxy"].(bool); ok {
+		obj.EnableCSIProxy = v
+	}
+
+	if v, ok := in["license_type"].(string); ok && len(v) > 0 {
+		obj.LicenseType = v
+	}
+	return obj
+}
+
+func expandAKSV3NodePool(p []interface{}) []*infrapb.Nodepool {
+	if len(p) == 0 || p[0] == nil {
+		return []*infrapb.Nodepool{}
+	}
+
+	out := make([]*infrapb.Nodepool, len(p))
+	outToSort := make([]infrapb.Nodepool, len(p))
+	for i := range p {
+		obj := infrapb.Nodepool{}
+		in := p[i].(map[string]interface{})
+
+		if v, ok := in["api_version"].(string); ok && len(v) > 0 {
+			obj.ApiVersion = v
+		}
+
+		if v, ok := in["name"].(string); ok && len(v) > 0 {
+			obj.Name = v
+		}
+
+		if v, ok := in["properties"].([]interface{}); ok && len(v) > 0 {
+			obj.Properties = expandAKSV3NodePoolProperties(v)
+		}
+
+		if v, ok := in["type"].(string); ok && len(v) > 0 {
+			obj.Type = v
+		}
+
+		if v, ok := in["location"].(string); ok && len(v) > 0 {
+			obj.Location = v
+		}
+		outToSort[i] = obj
+	}
+
+	sort.Sort(ByNodepoolNameV3(outToSort))
```

Comment:
@sbenny @poornima-rafay this sorting is becoming a problem in TF as TF thinks there is a change in the state. Is there a way to preserve the order that user gives?

Reply:
@kutumba-rafay I encountered this issue today. 
I am not sure how this is addressed in other tf yet. Probably need a solution at the DB level where data is stored and retrieved from in the same order. 
@sbenny please suggest whats the approach to take?